rdpng.c

```diff
@@ -102,8 +125,18 @@ start_input_png (j_compress_ptr cinfo, cjpeg_source_ptr sinfo)
     png_charp unused1 = NULL;
     int unused2 = 0;
     png_uint_32 proflen = 0;
-    if (png_get_iCCP(source->png_ptr, source->info_ptr, &unused1, &unused2, &profile, &proflen) && /* your libpng is out of date if you get a warning here */
-        profile && proflen) {
+    int has_profile = 0;
+
+    if (has_srgb_chunk) {
+        /* PNG can declare use of an sRGB profile without embedding an ICC file, but JPEG doesn't have such feature */
```

Comment:
Sure. Adobe’s not a problem, I think, as when files leave it (or originate in it) – a real profile is always embedded (unless user decides otherwise). Problem with colour described in just DCF is with cameras set to AdobeRGB (luckily: never a default). I blame Canon, Nikon et al. And then – Mozilla and Google for not recognising it (can’t claim not [knowing about it](https://bugzilla.mozilla.org/show_bug.cgi?id=16769#c30)). Can’t really expect open tools to be correct when both GM and IM are oblivious to it :sadpanda:.